pkg/compact/compact.go

```diff
@@ -271,6 +279,27 @@ func (c *Syncer) downloadMeta(ctx context.Context, id ulid.ULID) (*metadata.Meta
 	return &meta, nil
 }
 
+// removeIfMalformed removes a block from the bucket if that block does not have a meta file.
+// It is the responsibility of the caller to ensure that enough time has passed for the block to become consistent
+func (c *Syncer) removeIfMalformed(ctx context.Context, id ulid.ULID) bool {
```

Comment:
let's maybe be more accurate in naming:
```suggestion
func (c *Syncer) removeIfMetaMalformed(ctx context.Context, id ulid.ULID) bool {
```